custom_completions.go

```diff
@@ -176,7 +176,8 @@ func (c *Command) getCompletions(args []string) (*Command, []string, ShellCompDi
 	trimmedArgs := args[:len(args)-1]
 
 	// Find the real command for which completion must be performed
-	finalCmd, finalArgs, err := c.Root().Find(trimmedArgs)
+	// use Traverse here to parse local flags on parent commands
+	finalCmd, finalArgs, err := c.Root().Traverse(trimmedArgs)
```

Comment:
Could you add a test that fails if we use `Find()` when we shouldn't and when we use`Traverse()` when we shouldn't?

Reply:
The test should now fail when we would always use `Find()`.
I'm not sure how to make it fail when we use `Traverse()` though.

Reply:
This test should fail if we use `Traverse` instead of `Find`:
```
executeCommand(rootCmd, ShellCompNoDescRequestCmd, "--localroot", "value", "childCmd1", "")
```
Using `Find` (the passing test) will not return any completions because `--localroot` will be left on the command-line and `ParseFlags` will fail.
But using `Traverse` by mistake (the failing test) will return `childCmd2` as a completion because `--localroot` will have been removed from the command-line, which will allow `ParseFlags` to succeed.

Reply:
OK, thank you. I added the test.